### ISSUE TITLE
potential memory / resource leak on config reload

### ISSUE THREAD

**Issue by btvoidx** [lars-berger/GlazeWM#362]
I noticed that WM slows down considerably after a bunch of config reloads. A full restart fixes everything, of course. You might have to take a look at that. Thank you.

**Comment by KingMarti** [comment]
I have noticed this without the config reloads but if left running for a long time, the system becomes unresponsive until glazewm is killed and when it's re-opened it's all fine.

**Comment by akos-fekete** [comment]
> I have noticed this without the config reloads but if left running for a long time, the system becomes unresponsive until glazewm is killed and when it's re-opened it's all fine.

I also noticed this. I have to restart it once every 1-2 hours.

**Comment by K0K0S-TE** [comment]
I’m having the same issue. It’s strange because GlazeWM runs fine on my work laptop, but on my personal PC it starts lagging after about an hour. My PC is quite beefy, so performance shouldn’t be the problem. There’s nothing in the error log.

**Comment by KingMarti** [comment]
Going to take a look into this when I have some time this week, if I find anything I'll make a pull request (I'm a noob in rust so not committing to finding it) but it dosnt seem like a normal memory leak because for me at least task manager dosnt show the amount of ram being used increasing. 

System specs for reference: 

Ryzen 5 3600
ram: 40GB corsair vengeance 
GPU: 2060 Super

**Comment by K0K0S-TE** [comment]
@KingMarti I agree, doesn't look like memory leak to me too, my memory and cpu usage is very low, while pc is barely responsive.

I tried to check the error log file, but there is nothing there, and while running glazewm from terminal it doesn't log at all so I gave up - I was thinking about reinstalling windows.

My system specs for reference:
Ryzen 7 7800x3d
32GB G.Skill F5 6000MHz
4070 ti super
